packages/teleport/src/Kubes/KubeList/KubeList.tsx

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { sortBy } from 'lodash';
+import {
+  Column,
+  SortHeaderCell,
+  Cell,
+  TextCell,
+  SortTypes,
+} from 'design/DataTable';
+import { Label, Flex, ButtonBorder } from 'design';
+import Table from 'design/DataTable/Paged';
+import isMatch from 'design/utils/match';
+import { Kube } from 'teleport/services/kube';
+import InputSearch from 'teleport/components/InputSearch';
+import { AuthType } from 'teleport/services/user';
+import ConnectDialog from '../ConnectDialog';
+
+function KubeList(props: Props) {
+  const { kubes = [], pageSize = 20, username, authType } = props;
+
+  const [sortDir, setSortDir] = useState<Record<string, string>>({
+    name: SortTypes.DESC,
+  });
+  const [searchValue, setSearchValue] = useState('');
+  const [kubeConnectName, setKubeConnectName] = useState('');
+
+  function sortAndFilter(search) {
+    const filtered = kubes.filter(obj =>
+      isMatch(obj, search, {
+        searchableProps: ['name', 'tags'],
+        cb: searchAndFilterCb,
+      })
+    );
+
+    const columnKey = Object.getOwnPropertyNames(sortDir)[0];
+    const sorted = sortBy(filtered, columnKey);
+    if (sortDir[columnKey] === SortTypes.ASC) {
+      return sorted.reverse();
+    }
+
+    return sorted;
+  }
+
+  function onSortChange(columnKey: string, sortDir: string) {
+    setSortDir({ [columnKey]: sortDir });
+  }
+
+  const data = sortAndFilter(searchValue);
+
+  return (
+    <>
+      <Flex flex="0 0 auto" mb={4}>
```

Comment:
@amha19 
The search box does not really belong to this list, it needs to be moved 1 level up. One use case is that you can switch between multiple clusters and preserve the search value. With this, it will be reset each time a user changes a cluster.

It also a bit inconsistent with Nodes/Events/Etc where the search box does not disappear when content is being fetched.

Reply:
Initially, it was in the parent component, but later I moved it to the list for simplicity. 
now I've moved it back and updated the search value state.